aspnetcore/blazor/common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Shared/ContactForm.razor

```diff
@@ -0,0 +1,142 @@
+﻿@using Microsoft.AspNetCore.Components.Forms;
+
+<h1>@Mode Contact</h1>
+<hr />
+@if (Contact != null)
+{
+    <EditForm Model="@Contact"
+              OnInvalidSubmit="(async () => await HandleSubmitAsync(false))"
+              OnValidSubmit="(async () => await HandleSubmitAsync(true))">
+        <DataAnnotationsValidator />
+        <ValidationSummary />
+        <h2>🆔 Identification</h2>
+        <p>
+            <label>
+                First Name:
+                <InputText @bind-Value="Contact.FirstName" />
+                <ConcurrencyField Model="@Contact"
+                                  DbModel="@DbContact"
+                                  Property="obj => obj?.FirstName" />
```

Comment:
Can you explain a bit what this is for?

Reply:
For concurrency resolution. If the field was changed by another user, it will show a warning box with the changed value. That lets the current user decide if they want to revert changes or force an update.